src/main/java/core/basesyntax/domain/Fruit.java

```diff
@@ -0,0 +1,73 @@
+package core.basesyntax.domain;
+
+import java.util.Arrays;
+import java.util.Objects;
+
+public class Fruit {
+    private Operation operation;
+    private String name;
+    private int quantity;
+
+    public Fruit(String name, int quantity) {
+        this.name = name;
+        this.quantity = quantity;
+    }
+
+    public Fruit(Operation operation, String name, int quantity) {
+        this.operation = operation;
+        this.name = name;
+        this.quantity = quantity;
+    }
+
+    public Operation getOperation() {
+        return operation;
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    public int getQuantity() {
+        return quantity;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+        Fruit fruit = (Fruit) o;
+        return quantity == fruit.quantity
+                && operation == fruit.operation
+                && Objects.equals(name, fruit.name);
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(operation, name, quantity);
+    }
+
+    public enum Operation {
+        BALANCE("b"),
+        SUPPLY("s"),
+        PURCHASE("p"),
+        RETURN("r");
+
+        private String code;
+
+        Operation(String code) {
+            this.code = code;
+        }
+
+        public static Operation getByCode(String code) {
+            return Arrays.stream(Operation.values())
+                    .filter(operation -> operation.code.equals(code))
+                    .findFirst()
+                    .orElseThrow(() -> new RuntimeException("Invalid code" + code));
```

Comment:
The exception message in 'getByCode' method of the 'Operation' enum is missing a space between the words 'Invalid code' and the actual code value. It should be 'Invalid code ' + code to ensure proper readability of the error message.